### DISCUSSION TITLE
3D Preview

### DISCUSSION THREAD

**Discussion by matzmuda** [pimcore/pimcore#9861] [General]
Hello Everyone !
We are using Pimcore X CE, and we  are  creating  a new assets-portal for our 3d-designer. 
All assets are for 3d-max, soo we know that is not possible to make a preview.
But we want to save a duplicated assets in other format that will allows to use some JS library to make a preview of 3d-element.

My question is:
1. what format can you recommend , to make easy export from 3d-max to other format that will be readable for some Js-library ( to make preview) 
2. What format of 3d-assets can be handle via pimcore  by default?

ps. such kind of preview you can see on youtube here https://youtu.be/bJFlslL1wFI?t=309

Thank you

**Comment by aarongerig** [comment]
Hi @matzmuda
You could for example use Google's [<model-viewer>](https://modelviewer.dev) library to display 3D files. You'd have to convert your 3D files to GLB models and then upload those to Pimcore. Maybe you could even convert them on upload, so you don't have any duplicated assets nor manual work you have to do. Hope this works for 3D-max and helps you to accomplish what you are looking for.

If you manage to create that preview for 3D files, maybe it would also be a good idea to share the code as a community bundle. Would be a nice addition to the Pimcore platform!

**Reply by matzmuda** [reply]
@aarongerig  Thank you for your idea!

What I have also find is a three.js + .gLTF format.

https://www.youtube.com/watch?v=1TeMXIWRrqE 

I will wait for additional response , and then I will test all solution with out 3d-designer team. 
Thanks!
